### ISSUE TITLE
Test single cluster mode with external links

### ISSUE THREAD

**Issue by nrfox** [kiali/kiali#6233]
Ensure that the [existing experimental multi-cluster single control plane support](https://kiali.io/docs/features/multi-cluster/) still works as intended with the addition of the unified deployment.

**Comment by jmazzitelli** [comment]
@nrfox just so I'm clear - when you say "multi-cluster single control plane support" - this is referring to one Istio control plane with one Kiali installed, with the data plane across multiple clusters? (i.e. primery-remote)?

This issue with the Kiali links showing in the mesh page was really to show links to multiple Kialis in the multi-primary deployment scenario (so you can jump from one Kiali/control plane to another by clinking the link).

In short, I'm confused what this is asking for. :)

**Comment by nrfox** [comment]
> This issue with the Kiali links showing in the mesh page was really to show links to multiple Kialis in the multi-primary deployment scenario (so you can jump from one Kiali/control plane to another by clinking the link).

This issue is to test that this setup still works: multiple kialis with one per controlplane as opposed a single "unified" kiali.

**Comment by jmazzitelli** [comment]
> This issue is to test that this setup still works: multiple kialis with one per controlplane as opposed a single "unified" kiali.

OK, so testing with multi-primary mode, with kiali installed in each control plane.

I'm going to test with this:

```
hack/istio/multicluster/install-multi-primary.sh --client-exe kubectl -dorp podman --kiali-enabled true --single-kiali false
```
My predication is it won't work until I manually create the kiali remote secrets (that hack script with `--single-kiali` set to false won't create the remote secrets, if I read the code correct). I'm not sure if that is how we want it to work - we should not need the kiali remote secrets when installing in multi-primary/multi-kiali mode, but we also should not require permissions to read the istio secrets either (but that is probably going to need to be if we want these links to show up - the URLs come from data read from the secrets).

But let me play with this to see what really happens.

**Comment by nrfox** [comment]
> I'm not sure if that is how we want it to work - we should not need the kiali remote secrets when installing in multi-primary/multi-kiali mode, but we also should not require permissions to read the istio secrets either (but that is probably going to need to be if we want these links to show up - the URLs come from data read from the secrets).

So that's how this previously worked where Kiali could either read the istio secrets or you could create separate secrets manually. There's some knobs under `feature_flags.clustering` that control this: https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.kiali_feature_flags.clustering. We may decide we no longer want to support reading the istio secrets directly and only want to support mounting the secrets in to the Kiali pod.

**Comment by jmazzitelli** [comment]
with kiali remote cluster secrets created, this is what I see from the east Kiali:

![image](https://github.com/kiali/kiali/assets/2029470/111dba51-952e-4a18-b611-506007cdcb71)

**Comment by jmazzitelli** [comment]
@nrfox I added my comments here earlie. I just put you on as assignee since you also looked at this. Add your comments and feel free to close this issue if you think we don't need to do anything else.

**Comment by nrfox** [comment]
The links look good. Closing this as done.
